src/nightwatch/snapshot.js

```diff
@@ -0,0 +1,37 @@
+const Jimp = require('jimp');
+const fs = require('fs');
+const path = require('path');
+
+
+module.exports = function (browser, done) {
+  const testDirectory = path.dirname(module.parent.filename);
+  const currentTest = browser.currentTest;
+
+  browser.execute('document.documentElement.getAttribute("lang")', [], (locale) => {
+    const snapshotDirectory = path.join(testDirectory, '__snapshots__', locale.value || 'en');
+
+    browser.screenshot(false, (data) => {
+      Jimp.read(Buffer.from(data.value, 'base64')).then((snapshot) => {
+        const name = currentTest.name.replace(/\s+/g, '-');
+        const browserName = browser.capabilities.browserName;
+        const moduleName = currentTest.module.split('/').slice(-1)[0];
+        const fileName = `${moduleName}_${name}_${browserName}${snapshot.bitmap.width}.jpg`;
+        const filePath = path.join(snapshotDirectory, fileName);
+
+        // Found a baseline
+        if (fs.existsSync(filePath)) {
+          Jimp.read(filePath).then((baseline) => {
+            const diff = Jimp.diff(snapshot.autocrop(), baseline);
```

Comment:
It looks like the default difference threshold for Jimp is 0.1. IMO our threshold should probably be 0. It looks like there is a [bug](https://github.com/oliver-moran/jimp/issues/350) when defining the threshold as 0, but the workaround it using .01

Reply:
Or should threshold an optional value that will default to .01?

Reply:
we can do a perceptual hash compare if we want guaranteed identical. But yeah i'll update that to 0.1 and probably also allow it to be configurable

Reply:
Actually... i check the diff percent below and verify thats zero...

Reply:
Yeah, it looks like Jimp uses the pixelmatch package that takes this threshold value to return
the numDiffPixels which is used to calculate that percentage.